pkg/cmd/server/api/helpers.go

```diff
@@ -130,6 +130,10 @@ func GetMasterFileReferences(config *MasterConfig) []*string {
 	refs = append(refs, &config.EtcdClientInfo.ClientCert.KeyFile)
 	refs = append(refs, &config.EtcdClientInfo.CA)
 
+	refs = append(refs, &config.KubernetesMasterConfig.KubernetesEtcdClientInfo.ClientCert.CertFile)
```

Comment:
move these inside the `if config.KubernetesMasterConfig != nil` check below, and add an additional check to make sure `KubernetesEtcdClientInfo` is not nil